### PR TITLE
Disable create pool with non exit fee

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #4371 

## What is the purpose of the change

After discussing with @p0mvn, we decided to keep the current proto struct, and just not allow the creation of pools with non zero exit fees in the future.

## Brief Changelog

- add conditions when creating pool
- updated tests

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/poolmanager/create_pool_test.go [hunk 1]
```diff
@@ -155,6 +166,13 @@ func (suite *KeeperTestSuite) TestCreatePool() {
 			msg:                validConcentratedPoolMsg,
 			expectedModuleType: concentratedKeeperType,
 		},
+		{
+			name:               "pool with non zero exit fee - success",
```
Comment (p0mvn) [inline x/poolmanager/create_pool_test.go:170]:
```suggestion
			name:               "pool with non zero exit fee - error",
```

**Review by p0mvn — APPROVED** [review]
LGTM. Thanks.

Would like to request 2 items prior to merge:
1. Changelog entry
2. Comment in the proto file for `Pool`s`  `ExitFee` saying:
```
// N.B.: exit fee is disabled during pool creation in x/poolmanager. While old pools
// can maintain a non-zero fee. No new pool can be created with non-zero fee anymore
```

File: x/gamm/keeper/pool_service_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ import (
 
 var (
 	defaultSwapFee    = sdk.MustNewDecFromStr("0.025")
-	defaultExitFee    = sdk.MustNewDecFromStr("0.025")
+	defaultExitFee    = sdk.ZeroDec()
```
Comment (czarcas7ic) [inline x/gamm/keeper/pool_service_test.go:22]: nit: can we just define a `defaultZeroExitFee` in the keeper_test file as a global var and use it everywhere?

File: x/gamm/keeper/pool_service_test.go [hunk 3]
```diff
@@ -88,31 +88,31 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
...
-			name: "create a pool with negative exit fee",
+			name: "create a pool with non zero exit fee",
 			msg: balancer.NewMsgCreateBalancerPool(testAccount, balancer.PoolParams{
 				SwapFee: sdk.NewDecWithPrec(1, 2),
-				ExitFee: sdk.NewDecWithPrec(-1, 2),
+				ExitFee: sdk.NewDecWithPrec(1, 2),
 			}, defaultPoolAssets, defaultFutureGovernor),
 			emptySender: false,
 			expectPass:  false,
 		}, {
```
Comment (czarcas7ic) [inline x/gamm/keeper/pool_service_test.go:103]: imo we should test both a negative and a non zero exit fee

**Review by czarcas7ic — APPROVED** [review]

**Comment by hieuvubk** [comment]
All cmts were addressed, thank u so much

**Pull request merged**